.github/workflows/release.yml

```diff
@@ -0,0 +1,167 @@
+name: Release
+
+env:
+  GO111MODULE: "on"
+  GOPATH: "/home/runner/work/armada/armada/go"
+  GOCACHE: "/home/runner/.cache/go-build"
+  GOVERSION: "1.16"
+
+on: 
+  push:
+    tags: 
+      - 'v*'
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    env:
+      DOCKERHUB_PASS: ${{ secrets.DOCKERTOKEN }}
+      DOCKERHUB_USER: ${{ secrets.DOCKERHUB_USER }}
+    steps:
+      - name: Checkout code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GOVERSION }}
+
+      - name: Build armadactl release artifacts
+        run: |
+          TAG=${GITHUB_REF##*/}
+          make build-armadactl-release RELEASE_VERSION=${TAG}
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - name: Archive production artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: dist
+          path: ${{ env.GOPATH }}/src/github.com/G-Research/armada/dist/*
+
+      #### Publish release
+      - name: Upload artifacts to Github release
+        uses: ncipollo/release-action@v1.8.6
+        with:
+          artifacts: "${{ env.GOPATH }}/src/github.com/G-Research/armada/dist/*"
+          token: ${{ secrets.GITHUB_TOKEN }}
+          tag: ${{ github.event.release.tag_name }} 
+
+      - name: Push Release Image
+        run: |
+          if [ -z "${DOCKERHUB_USER}" ]
```

Comment:
@ljubon do you think we can use something similar as here to check for the fork status: https://github.com/pavlovic-ivan/ILGPU/blob/master/.github/workflows/runners.yaml#L14

Reply:
I am not sure about checking if it's a fork or not, but the logic for checking if `DOCKERHUB_USER` is set should be fine to have it in job condition
The valid case could be that someone wants to use this and push this to his repo, that's still valid as it's open-source, still nothing is stoping anyone to remove this lock once it's forked and push wherever he wants
In the end, we want to protect `DOCKERTOKEN` and `DOCKERHUB_USER` 
I'll push this change, but I need to test if this implementation exposes secrets (hope not 🤞 )